<a href="https://colab.research.google.com/github/SAULIZA/Proyecto/blob/main/Proyecto2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Proyecto2 - Parte1(base)
El segundo proyecto tendrá más libertad que el primero. Esto es porque queremos que tengan un proyecto en sus portafolios que les interese o se relacione con la industria en la que quieren trabajar.

La tarea para esta semana es proponer dos posibles conjuntos de datos que les gustaría trabajar para el proyecto 2.  

Escogerán el conjunto de datos de primera opción, y un conjunto de datos de reserva en caso de que el primero propuesto no se apruebe.  

Estos datos pueden proceder de cualquier fuente y ser sobre cualquier tema con estas limitaciones:

* los datos deben estar disponibles para su uso (es su responsabilidad garantizar que la licencia indique que pueden utilizarlos)
* los datos deben ser adecuados para un ambiente profesional
* los datos NO deben contener información personal
* los datos NO deben ser de un conjunto de datos usados en alguna tarea o clase del curso

Asegúrense de seleccionar un conjunto de datos con el que sea razonable trabajar en el tiempo que nos queda. Piensen qué preguntas podrían responder con el conjunto de datos que seleccionen. 

Deben proponer dos conjuntos de datos que tengan cada uno un componente de aprendizaje supervisado. Pueden elegir un problema de regresión o clasificación para cada conjunto de datos propuesto.  

**Para esta tarea:**

Creen un cuaderno de Colab donde hayan subido y mostrado el .head() de cada uno de los conjuntos de datos.  Para cada uno de los conjuntos de datos propuestos, respondan las siguientes preguntas:

## Conjunto de datos 1:

**Datos meteorológicos de la ciudad de Quito**

In [3]:
from google.colab import drive
drive.mount("/content/drive")

Mounted at /content/drive


In [1]:
import pandas as pd

1. Fuente de datos
* La fuente de los datos meteorológicos de la ciudad de Quito provienen de la página web de la secretaría del ambiente de Quito (http://www.quitoambiente.gob.ec/index.php/politicas-y-planeacion-ambiental/red-de-monitoreo)

In [5]:
df_co = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/CO.xlsx")
#df_dir = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/DIR.xlsx")
df_hum = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/HUM.xlsx")
df_iuv = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/IUV.xls")
df_llu = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/LLU.xlsx")
df_no2 = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/NO2.xlsx")
df_o3 = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/O3.xlsx")
df_pm10 = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/PM10.xlsx")
df_pm2_5 = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/PM2.5.xlsx")
df_pre = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/PRE.xlsx")
df_rs = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/RS.xlsx")
df_s02 = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/SO2.xlsx")
df_tmp = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/TMP.xlsx")
df_vel = pd.read_excel("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/datos meteorológicos/VEL.xlsx")

2. Breve descripción de datos:
* Se tienen datos meteorológicos de la ciudad de quito, de diversos sectores de la ciudad. Cada conjunto de datos contiene un parámetro específico, como por ejemplo df_iuv, contiene datos de índice ultravioleta medidos en distintos sectores de la ciudad desde el año 2004 hasta la actualidad.

In [6]:
df_co.head()

Unnamed: 0 BELISARIO CARAPUNGO CENTRO COTOCOLLAO EL CAMAL GUAMANI  \
0      FECHA \ UNIDAD      mg/m3     mg/m3  mg/m3      mg/m3    mg/m3   mg/m3   
1  2004-01-01 00:00:00      7.42       NaN   4.33        NaN     3.96     NaN   
2  2004-01-01 01:00:00      7.96       NaN   4.45        NaN     3.49     NaN   
3  2004-01-01 02:00:00      8.42       NaN   3.53        NaN     1.78     NaN   
4  2004-01-01 03:00:00      9.06       NaN   3.36        NaN     1.29     NaN   

  LOS CHILLOS TUMBACO CONDADO TURUBAMBA CHILLOGALLO  
0       mg/m3   mg/m3   mg/m3     mg/m3       mg/m3  
1         NaN     NaN    3.49       4.3         NaN  
2         NaN     NaN    3.73      4.01         NaN  
3         NaN     NaN    5.36      3.75         NaN  
4         NaN     NaN    4.42      3.23         NaN

In [7]:
df_hum.head()

Unnamed: 0 Belisario Carapungo Centro Cotocollao ElCamal Guamaní  \
0       FECHA \ UNIDAD         %         %      %          %       %       %   
1  2004-01-01 00:00:00     98.06     74.78    NaN      98.49   76.89     NaN   
2  2004-01-01 01:00:00     98.47     72.42    NaN        100   75.94     NaN   
3  2004-01-01 02:00:00     98.65     74.43    NaN        100   76.08     NaN   
4  2004-01-01 03:00:00     99.03     76.16    NaN        100   78.94     NaN   

  LosChillos SanAntonio Tumbaco  
0          %          %       %  
1      93.17        NaN   98.56  
2      94.62        NaN   99.61  
3      97.13        NaN   99.14  
4        100        NaN   98.45

In [8]:
df_iuv.head()

Unnamed: 0 Centro Cotocollao Guamani Jipijapa
0       FECHA \ UNIDAD    IUV        IUV     IUV      IUV
1  2017-08-01 00:00:00    NaN          0       0      NaN
2  2017-08-01 01:00:00    NaN          0       0      NaN
3  2017-08-01 02:00:00    NaN          0       0      NaN
4  2017-08-01 03:00:00    NaN          0       0      NaN

In [9]:
df_s02.head()

Unnamed: 0 BELISARIO CARAPUNGO CENTRO COTOCOLLAO EL CAMAL GUAMANI  \
0       Fecha \ Unidad     ug/m3     ug/m3  ug/m3      ug/m3    ug/m3   ug/m3   
1  2004-01-01 00:00:00       NaN       NaN   9.58        NaN      NaN     NaN   
2  2004-01-01 01:00:00       NaN       NaN   9.43        NaN      NaN     NaN   
3  2004-01-01 02:00:00       NaN       NaN   7.49        NaN      NaN     NaN   
4  2004-01-01 03:00:00       NaN       NaN   5.31        NaN      NaN     NaN   

  LOS CHILLOS TUMBACO CONDADO TURUBAMBA CHILLOGALLO  
0       ug/m3   ug/m3   ug/m3     ug/m3       ug/m3  
1         NaN     NaN   42.14     39.43         NaN  
2         NaN     NaN    66.1     21.11         NaN  
3         NaN     NaN   68.52     10.49         NaN  
4         NaN     NaN   57.62      6.23         NaN

In [10]:
df_tmp.head()

Unnamed: 0 Belisario Carapungo Centro Cotocollao ElCamal Guamaní  \
0       Fecha \ Unidad        °C        °C     °C         °C      °C      °C   
1  2004-01-01 00:00:00      9.93     12.73    NaN      10.41   12.57     NaN   
2  2004-01-01 01:00:00      9.14     12.58    NaN       9.76   12.16     NaN   
3  2004-01-01 02:00:00      8.71     12.12    NaN       9.49   11.87     NaN   
4  2004-01-01 03:00:00      8.63     11.94    NaN       9.44   11.38     NaN   

  LosChillos SanAntonio Tumbaco  Unnamed: 10  Unnamed: 11  Unnamed: 12  \
0         °C         °C      °C          NaN          NaN          NaN   
1       9.95        NaN   12.36          NaN          NaN          NaN   
2       8.77        NaN   11.87          NaN          NaN          NaN   
3       8.35        NaN   11.82          NaN          NaN          NaN   
4       8.01        NaN   11.62          NaN          NaN          NaN   

  Unnamed: 13  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN

3. ¿Cuál es el objetivo?
* El objetivo es usar los datos meteorológicos de Quito para, encontrar como ha ido variando los distintos parámetros meteorológicos con el tiempo. Averiguar que tan predecible es el clima de Quito si sabemos algunos parámetros de antemano. Averiguar la magnitud y variación del calentamiento global en los últimos 20 años en la ciudad. Así mismo podríamos averiguar si la contaminación en la ciudad incrementa con el tiempo, y cuales son los sectores más afectados. Estos serían algunos estudios que podríamos hacer.

4. ¿Este es un problema de clasificación o regresión?
* Es un problema de regresión.

5. ¿Cuántas características hay?
Tenemos 14 características, probablemente se reduzca a 13 ya que df_dir (dirección del viento) no sería de mucha utilidad.

6. ¿Cuántas filas de datos hay?


In [12]:
df_co.shape

(167305, 12)

In [13]:
df_hum.shape

(167305, 10)

In [14]:
df_iuv.shape

(48241, 5)

In [15]:
df_pre.shape

(167305, 10)

El número de filas depende del parámetros, en algunos casos son de 167305 filas y otros 48241 filas, depende del intervalo de tiempo de registro de los distintos parámetros.

7. ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

In [17]:
df_co.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 167305 entries, 0 to 167304
Data columns (total 12 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   Unnamed: 0   167305 non-null  object
 1   BELISARIO    162741 non-null  object
 2   CARAPUNGO    153184 non-null  object
 3   CENTRO       160146 non-null  object
 4   COTOCOLLAO   153173 non-null  object
 5   EL CAMAL     142986 non-null  object
 6   GUAMANI      152237 non-null  object
 7   LOS CHILLOS  76540 non-null   object
 8   TUMBACO      30155 non-null   object
 9   CONDADO      9703 non-null    object
 10  TURUBAMBA    9870 non-null    object
 11  CHILLOGALLO  1 non-null       object
dtypes: object(12)
memory usage: 15.3+ MB


In [18]:
df_pre.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 167305 entries, 0 to 167304
Data columns (total 10 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   Unnamed: 0  167305 non-null  object
 1   Belisario   164929 non-null  object
 2   Carapungo   163173 non-null  object
 3   Centro      36141 non-null   object
 4   Cotocollao  164453 non-null  object
 5   ElCamal     144771 non-null  object
 6   Guamaní     52137 non-null   object
 7   LosChillos  164485 non-null  object
 8   SanAntonio  50861 non-null   object
 9   Tumbaco     161065 non-null  object
dtypes: object(10)
memory usage: 12.8+ MB


In [19]:
df_iuv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48241 entries, 0 to 48240
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Unnamed: 0  48241 non-null  object
 1   Centro      36573 non-null  object
 2   Cotocollao  40330 non-null  object
 3   Guamani     25326 non-null  object
 4   Jipijapa    35425 non-null  object
dtypes: object(5)
memory usage: 1.8+ MB


En la limpieza de datos nos enfrentamos a celdas vacías y que los parámetros no tienen registros en todos los sectores de la ciudad, más bien el regitro de los parámetros depende del lugar, hay lugares que no tienen registros de ciertos parámetros.

## Conjunto de datos 2
**Ventas y compras mensualmente**

1. Fuente de datos
* Los datos de ventas y compras mensualmente, por provincia y actividad económica provienen de los datasets del sri (https://www.sri.gob.ec/datasets)

2. Breve descripción de datos
* Los datos de ventas y compras, son datos de empresas registradas en el SRI. Los datos se encuentran en archivos por año, en este caso se presentan los datos del año 2017.

In [44]:
df_ventas = pd.read_csv("/content/drive/MyDrive/Codificación de Dojo/Proyecto2/ventas_sri/sri_ventas_2017.csv", sep=";",encoding='ISO-8859-1')

In [45]:
df_ventas.head()

AÑO  MES CODIGO_SECTOR_N1   PROVINCIA      CANTON Ventas netas tarifa 12%  \
0  2017   11                C   PICHINCHA       QUITO            674109759,63   
1  2017    8                C      MANABI  PORTOVIEJO              3531004,05   
2  2017    5                L  TUNGURAHUA      AMBATO               3375473,7   
3  2017   11                P      GUAYAS   GUAYAQUIL              4001576,28   
4  2017    1                G      MANABI  PEDERNALES              2779580,87   

  Ventas netas tarifa 0% EXPORTACIONES Compras netas tarifa 12%  \
0           157733836,78   48962950,83             293915481,96   
1             1817252,77     931147,88               2894980,12   
2             1275353,35             0               1726147,81   
3            34262278,02             0              16130416,09   
4             3359265,24    1356437,73               3092834,67   

  Compras netas tarifa 0% IMPORTACIONES COMPRAS_RISE TOTAL_COMPRAS  \
0            126863353,05  229887459,47   1848572,32  663526526,28   
1              1779175,07      15828,28     72562,35    4762545,82   
2               512450,62       8830,71     22338,29    2269767,43   
3              4974618,79      28440,65    170121,94   21303887,37   
4              4144722,44       94202,1     35147,97    7366907,18   

   TOTAL_VENTAS  
0  880806547,24  
1     6279404,7  
2    4650827,05  
3    38263854,3  
4    7495283,84

3. ¿Cuál es el objetivo?
* El objetivo es predecir el total de ventas de las distintas empresas conociendo algúnas características como las compras, exportaciones, importaciones entre otras variables.

4. ¿Este es un problema de clasificación o regresión?
* Este es un problema de regresión

5. ¿Cuántas características hay?
* Hay 13 características

In [46]:
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50313 entries, 0 to 50312
Data columns (total 14 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   AÑO                       50313 non-null  int64 
 1   MES                       50313 non-null  int64 
 2   CODIGO_SECTOR_N1          50313 non-null  object
 3   PROVINCIA                 50313 non-null  object
 4   CANTON                    50313 non-null  object
 5   Ventas netas tarifa 12%   50313 non-null  object
 6   Ventas netas tarifa 0%    50313 non-null  object
 7   EXPORTACIONES             42568 non-null  object
 8   Compras netas tarifa 12%  50313 non-null  object
 9   Compras netas tarifa 0%   50313 non-null  object
 10  IMPORTACIONES             42568 non-null  object
 11  COMPRAS_RISE              50313 non-null  object
 12  TOTAL_COMPRAS             50313 non-null  object
 13  TOTAL_VENTAS              50313 non-null  object
dtypes: int64(2), object(12

6. ¿Cuántas filas de datos hay?
* Hay 50313 filas


7. ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?
* El reto más grande en la limpieza de datos son las celdas vacias que se encuentran en dos columnas.